File: server/WebApi/Repository/Entities/CurrentUserLevel.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Repository.Entities
{
    public class CurrentUserLevel
    {
        [Key]
        public int Id { get; set; }
        [ForeignKey("User")]
        public int UserId { get; set; }
        public string GrammarLevel { get; set; }
        public string VocabularyLevel { get; set; }
        public string ReadingLevel { get; set; }
        public DateTime DateUpdated { get; set; }
    }
}

File: client/english-project/src/types/currentUserLevel.ts
export type LevelType = 'Beginner' | 'Intermediate' | 'Advanced';

export type CurrentUserLevel = {
    userId: number;
    grammarLevel: LevelType;
    vocabularyLevel: LevelType;
    readingLevel: LevelType;
    dateUpdated: Date;
}

export const calculateLevel = (percentage: number): LevelType => {
    if (percentage < 50) return 'Beginner';
    if (percentage < 80) return 'Intermediate';
    return 'Advanced';
};